tests/unit/versioning/VersionID.spec.js

```diff
@@ -9,83 +10,124 @@ function randkey(length) {
+function generateRandomVIDs(count) {
     const vids = Array(count).fill(null);
     for (let i = 0; i < count; i++) {
         vids[i] = VID.generateVersionId(randkey(15), 'PARIS');
     }
     process.env.VID_CRYPTO_PASSWORD = randkey(64);
+    return vids;
+}
+
+const count = 1000000;
+
+describe('test generating versionIds', () => {
+    describe('legaxy hex encoding', () => {
+        env.S3_VERSION_ID_ENCODING_TYPE = 'hex';
+        const vids = generateRandomVIDs(count);
 
-    it('sorted in reversed chronological and alphabetical order', () => {
-        for (let i = 0; i < count; i++) {
-            if (i !== 0) {
-                assert(vids[i - 1] > vids[i],
-                    'previous VersionID is higher than its next');
+        it('sorted in reversed chronological and alphabetical order', () => {
+            for (let i = 0; i < count; i++) {
+                if (i !== 0) {
+                    assert(vids[i - 1] > vids[i],
+                        'previous VersionID is higher than its next');
+                }
             }
-        }
-    });
+        });
 
-    // nodejs 10 no longer returns error for non-hex string versionIds
-    it.skip('should return error decoding non-hex string versionIds', () => {
-        const encoded = vids.map(vid => VID.hexEncode(vid));
-        const decoded = encoded.map(vid => VID.hexDecode(`${vid}foo`));
-        decoded.forEach(result => assert(result instanceof Error));
-    });
+        // nodejs 10 no longer returns error for non-hex string versionIds
+        it.skip('should return error decoding non-hex string versionIds', () => {
+            const encoded = vids.map(vid => VID.hexEncode(vid));
+            const decoded = encoded.map(vid => VID.hexDecode(`${vid}foo`));
+            decoded.forEach(result => assert(result instanceof Error));
+        });
 
-    it('should encode and decode versionIds', () => {
-        const encoded = vids.map(vid => VID.hexEncode(vid));
-        const decoded = encoded.map(vid => VID.hexDecode(vid));
-        assert.strictEqual(vids.length, count);
-        assert.deepStrictEqual(vids, decoded);
-    });
+        it('should encode and decode versionIds', () => {
+            const encoded = vids.map(vid => VID.hexEncode(vid));
+            const decoded = encoded.map(vid => VID.hexDecode(vid));
+            assert.strictEqual(vids.length, count);
+            assert.deepStrictEqual(vids, decoded);
+        });
 
-    it('simple base62 version test', () => {
-        const vid = '98376906954349999999RG001  145.20.5';
-        const encoded = VID.base62Encode(vid);
-        assert.strictEqual(encoded, 'aJLWKz4Ko9IjBBgXKj5KQT2G9UHv0g7P');
-        const decoded = VID.base62Decode(encoded);
-        assert.strictEqual(vid, decoded);
-    });
+        it('should encode and decode correctly with legacy format', () => {
+            const encoded = vids.map(vid => VID.encode(vid));
+            const decoded = encoded.map(vid => VID.decode(vid));
```

Comment:
```suggestion
            const encoded = vids.map(VID.encode);
            const decoded = encoded.map(VID.decode);
```
Does this work?

Reply:
I'm trying to find the es-lint rule for this. Do you have a link? We don't enforce it right now though and I'm not sure if we should be introducing linting rules in a random pr.

Reply:
This is a linting rule? I was just suggesting it to pass the function directly instead of passing an inline function for the same thing. Not a big deal either way.